client/src/components/QuestionButton/QuestionButton.js

```diff
@@ -3,12 +3,12 @@ import Button from "react-bootstrap/Button";
 
 function QuestionButton(props) {
   const handleClick = () => {
-    props.onQuestionChange();
-    props.onDisabled();
+    props.onQuestionChange(props.question.id);
```

Comment:
I changed the logic of the question button disabled state to be tied to a map of what questions had been answered. That makes "is this button disabled?" able to be derived from other state, so we don't actually need to manually set disabled status.

If you're able to derive some state from some other state, that means you don't need to track it. More state = more bugs, so try to minimize state as much as possible.